neverbleed.c

```diff
@@ -548,10 +597,38 @@ static size_t daemon_set_rsa(RSA *rsa)
     return index;
 }
 
+// sets up file descriptor and calls OpenSSL's `ASYNC_pause_job`
+// if OpenSSL async is not available, this function is a no-op
+static int async_pause(int fd)
+{
+#ifdef NEVERBLEED_OPENSSL_HAVE_ASYNC
+    ASYNC_JOB *job;
+
+    if ((job = ASYNC_get_current_job()) != NULL) {
+        ASYNC_WAIT_CTX *waitctx = ASYNC_get_wait_ctx(job);
+
+        size_t numfds;
+        assert(ASYNC_WAIT_CTX_get_all_fds(waitctx, NULL, &numfds) && numfds == 0);
+        if (!ASYNC_WAIT_CTX_set_wait_fd(waitctx, "neverbleed", fd, NULL, NULL)) {
+            fprintf(stderr, "could not set async fd\n");
+            return -1;
+        }
+        ASYNC_pause_job();
+        if (!ASYNC_WAIT_CTX_clear_fd(waitctx, "neverbleed")) {
+            fprintf(stderr, "could not clear async fd\n");
+            return -1;
```

Comment:
Can we continue operating reliably after this failure, or is it worth trying so?

Unless the answer is yes for both questions, I wonder if we should simply change the return type of `async_pause` to `void`, using `dief` when detecting errors that should not happen.